python_modules/libraries/dagster-dbt/dagster_dbt/asset_utils.py

```diff
@@ -680,255 +680,215 @@ def is_non_asset_node(dbt_resource_props: Mapping[str, Any]):
+            if not child_unique_id.startswith("test"):
+                continue
 
-            metadata = _attach_sql_model_code_reference(
-                existing_metadata=metadata,
-                dbt_resource_props=dbt_resource_props,
-                project=project,
-            )
-
-        spec = AssetSpec(
-            key=asset_key,
-            description=dagster_dbt_translator.get_description(dbt_resource_props),
-            metadata=metadata,
-            owners=dagster_dbt_translator.get_owners(
-                {
-                    **dbt_resource_props,
-                    **({"group": dbt_group_resource_props} if dbt_group_resource_props else {}),
-                }
-            ),
-            tags={
-                **build_kind_tag("dbt"),
-                **(build_kind_tag(dbt_adapter_type) if dbt_adapter_type else {}),
-                **dagster_dbt_translator.get_tags(dbt_resource_props),
-            },
-            group_name=dagster_dbt_translator.get_group_name(dbt_resource_props),
-            code_version=dagster_dbt_translator.get_code_version(dbt_resource_props),
-            freshness_policy=dagster_dbt_translator.get_freshness_policy(dbt_resource_props),
-            automation_condition=dagster_dbt_translator.get_automation_condition(
-                dbt_resource_props
-            ),
-            partitions_def=dagster_dbt_translator.get_partitions_def(dbt_resource_props),
-        )
-
-        outs[output_name] = AssetOut.from_spec(
-            spec=spec,
-            dagster_type=Nothing,
-            is_required=False,
-            io_manager_key=io_manager_key,
-        )
-
-        test_unique_ids = [
-            child_unique_id
-            for child_unique_id in manifest["child_map"][unique_id]
-            if child_unique_id.startswith("test")
-        ]
-        for test_unique_id in test_unique_ids:
             check_spec = default_asset_check_fn(
-                manifest,
-                dbt_resource_props_by_dbt_unique_id,
-                dagster_dbt_translator,
-                asset_key,
-                test_unique_id,
+                manifest, dbt_nodes, translator, spec.key, child_unique_id
             )
             if check_spec:
-                check_specs_by_key[check_spec.key] = check_spec
+                check_specs.append(check_spec)
 
-        # Translate parent unique ids to dependencies
-        output_internal_deps = internal_asset_deps.setdefault(output_name, set())
-        for parent_unique_id in parent_unique_ids:
-            dbt_parent_resource_props = dbt_resource_props_by_dbt_unique_id[parent_unique_id]
-            parent_asset_key = dagster_dbt_translator.get_asset_key(dbt_parent_resource_props)
-            parent_partition_mapping = dagster_dbt_translator.get_partition_mapping(
-                dbt_resource_props,
-                dbt_parent_resource_props=dbt_parent_resource_props,
-            )
+        # TODO: better way to handle this
```

Comment:
linear ticket plox

Reply:
thought about it and this sort of thing is totally fine right now and so no work actually needs to be done, it's more contingent on future refactors potentially breaking this pattern.

updated the comment to be a nicer note for whoever next touches this